nixops/deployment.py

```diff
@@ -138,22 +143,24 @@ def active_resources(self) -> Dict[str, nixops.resources.GenericResourceState]:
         return {n: r for n, r in self.resources.items() if not r.obsolete}
 
     def get_typed_resource(
-        self, name: str, type: str
-    ) -> nixops.resources.GenericResourceState:
+        self, name: str, type_name: str, type: Type[TypedResource]
+    ) -> TypedResource:
         res = self.active_resources.get(name, None)
         if not res:
             raise Exception("resource ‘{0}’ does not exist".format(name))
-        if res.get_type() != type:
-            raise Exception("resource ‘{0}’ is not of type ‘{1}’".format(name, type))
-        return res
+        if res.get_type() != type_name:
+            raise Exception(
+                "resource ‘{0}’ is not of type ‘{1}’".format(name, type_name)
+            )
+        return res  # type:ignore
 
-    def get_machine(self, name: str) -> nixops.resources.GenericResourceState:
+    def get_machine(self, name: str, type: Type[TypedResource]) -> TypedResource:
         res = self.active_resources.get(name, None)
```

Comment:
I wonder if we should have a get_generic_machine/resource, and a get_typed_machine/resource. What do you think?

Reply:
I was considering returning to
```python 
def get_machine(self, name: str) -> nixops.resources.GenericResourceState:
```
and let the caller use [cast](https://docs.python.org/3/library/typing.html#typing.cast) instead.

This results in exactly the same type safety as 
```python 
def get_machine(self, name: str, type: Type[TypedResource]) -> TypedResource:
```
with only some slight inconvenience for the caller.

Wdyt?

Reply:
I like the second option much better. It is really awkward to use in other libraries: https://github.com/input-output-hk/nixops-packet/pull/21/files#diff-78f3655b013910dfdb430d53dae85085R11-R18 this feels like something that should definitely be implemented in core, to provide a safe API.